src/ray/core_worker/core_worker.cc

```diff
@@ -1174,8 +1174,77 @@ Status CoreWorker::SetResource(const std::string &resource_name, const double ca
   return local_raylet_client_->SetResource(resource_name, capacity, client_id);
 }
 
-Status CoreWorker::ForceSpillObjects(const std::vector<ObjectID> &object_ids) {
-  return local_raylet_client_->ForceSpillObjects(object_ids);
+Status CoreWorker::SpillObjects(const std::vector<ObjectID> &object_ids) {
+  absl::Mutex mutex;
+  auto num_remaining = std::make_shared<size_t>(object_ids.size());
+  auto final_status = std::make_shared<Status>();
+  auto ready_promise = std::make_shared<std::promise<void>>(std::promise<void>());
+
+  for (const auto &object_id : object_ids) {
+    RAY_LOG(DEBUG) << "Requesting spill for object " << object_id;
+    AddLocalReference(object_id, "<temporary (get object status)>");
```

Comment:
I don't think we need to bother with pinning.

Reply:
We need to do it to make sure we add the `GetAsync` callback while the object is in scope. Otherwise we can cause a memory leak where the callback never gets called.

Reply:
Can you add a comment to this affect?